dfget/util/log_test.go

```diff
@@ -123,15 +113,17 @@ func (suite *DFGetUtilSuite) TestAddConsoleLog(c *check.C) {
 func tempFileAndLogger(level string, sign string) (*logrus.Logger, *os.File, *bufio.Reader, error) {
 	tmpPath := "/tmp"
 	tmpFile, err := ioutil.TempFile(tmpPath, "dfget_test")
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
 	tmpFileName := strings.TrimLeft(tmpFile.Name(), "/tmp/")
 	r := bufio.NewReader(tmpFile)
-	logger := CreateLogger(tmpPath, tmpFileName, level, sign)
+	logger, err := CreateLogger(tmpPath, tmpFileName, level, sign)
 	return logger, tmpFile, r, err
 }
 
-func cleanTempFile(file *os.File, err error) {
-	if err == nil {
-		file.Close()
-		os.Remove(file.Name())
-	}
+func cleanTempFile(file *os.File) {
```

Comment:
The file is `nil` when an error occurred.
It's better to retain the parameter `err` even though you use `c.Assert` to check it before calling this function. It requires less pre-conditions and keeps up robustness for other callers to use it conveniently.